RealmModelGenerator/EntitiesViewController.swift

```diff
@@ -0,0 +1,69 @@
+//
+//  EntitieViewController.swift
+//  RealmModelGenerator
+//
+//  Created by Zhaolong Zhong on 3/21/16.
+//  Copyright © 2016 QuarkWorks. All rights reserved.
+//
+
+import Cocoa
+
+
+class EntitiesViewController: NSViewController, EntitiesViewDelegate, EntitiesViewDataSource {
+    static let TAG = NSStringFromClass(EntitiesViewController)
+    
+    @IBOutlet weak var entitiesView: EntitiesView!
+    
+    private var schema:Schema = Schema()
+    private var model: Model?
+    private var entities:[Entity] = []
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // Do any additional setup after loading the view.
+    }
+    
+    override func viewWillAppear() {
+        entitiesView.delegate = self
+        entitiesView.dataSource = self
+        
+        if let currentModel = schema.getCurrentModel() {
+            model = currentModel
+        } else {
+            model = schema.createModel()
+        }
+
+        entities = model!.entities
+    }
+    
+    //MARK: - EntitiesViewDelegate
+    func entitiesView(entitiesView: NSTableView, viewForTableColmn tableColumn: NSTableColumn?, row: Int) -> NSView? {
+        let cell = entitiesView.makeViewWithIdentifier(TitleCell.IDENTIFIER, owner: nil) as! TitleCell
+        cell.title = entities[row].name
+        return cell
+    }
+    
+    func entitiesView(entitiesView:EntitiesView, selectionDidChange index:Int) {
+        //TODO: notify attribute, relationship, and entity detail data source change
+    }
+    
+    func numberOfRowsInEntitiesView(entitiesView: EntitiesView) -> Int {
+        return entities.count
+        
+    }
+    
+    func addEntityInEntitiesView(entitiesView: EntitiesView) {
+        entities.append(model!.createEntity())
+    }
+    
+    func removeEntityInEntitiesView(entitiesView: EntitiesView, index:Int) {
+        if (index >= 0 && index < entities.count) {
+            model!.removeEntity(entities[index])
+            entities.removeAtIndex(index)
+        }
+    }
+    
+    func setSchema(schema: Schema) {
```

Comment:
try to use the swift property observers if you want to handle set and get, basically this is handled from the var schema:Schema? up above.